src/anomalib/deploy/inferencers/torch_inferencer.py

```diff
@@ -128,21 +130,39 @@ def post_process(self, predictions: Tensor, metadata: dict | DictConfig | None =
         if metadata is None:
             metadata = self.metadata
 
+        # Some models return a Tensor while others return a list or dictionary. Handle both cases.
+        # TODO: This is a temporary fix. We will wrap this post-processing stage within the model's forward pass.
+
+        # Case I: Predictions could be a tensor.
         if isinstance(predictions, Tensor):
             anomaly_map = predictions.detach().cpu().numpy()
             pred_score = anomaly_map.reshape(-1).max()
-        else:
-            # NOTE: Patchcore `forward`` returns heatmap and score.
-            #   We need to add the following check to ensure the variables
-            #   are properly assigned. Without this check, the code
-            #   throws an error regarding type mismatch torch vs np.
+
+        # Case II: Predictions could be a dictionary of tensors.
+        elif isinstance(predictions, dict):
+            if "anomaly_map" in predictions:
+                anomaly_map = predictions["anomaly_map"].detach().cpu().numpy()
+            else:
+                raise KeyError("``anomaly_map`` not found in the predictions.")
+
+            if "pred_score" in predictions:
+                pred_score = predictions["pred_score"].detach().cpu().numpy()
+            else:
+                raise KeyError("``pred_score`` not found in the predictions.")
```

Comment:
I think this might be problematic when model returns just an anomaly map in a dict (Efficient AD). In that case anomaly score is taken as a max value in anomaly map, same as in case 1.